biodatasets/n2c2_2018_track1/n2c2_2018_track1.py

```diff
@@ -0,0 +1,282 @@
+}
+"""
+
+_DATASETNAME = "n2c2_2018_track1"
+
+_DESCRIPTION = """\
+Track 1 of the 2018 National NLP Clinical Challenges shared tasks focused
+on identifying which patients in a corpus of longitudinal medical records
+meet and do not meet identified selection criteria.
+
+This shared task aimed to determine whether NLP systems could be trained to identify if patients met or did not meet
+a set of selection criteria taken from real clinical trials. The selected criteria required measurement detection (
+“Any HbA1c value between 6.5 and 9.5%”), inference (“Use of aspirin to prevent myocardial infarction”),
+temporal reasoning (“Diagnosis of ketoacidosis in the past year”), and expert judgment to assess (“Major
+diabetes-related complication”). For the corpus, we used the dataset of American English, longitudinal clinical
+narratives from the 2014 i2b2/UTHealth shared task 4.
+
+The final selected 13 selection criteria are as follows:
+1. DRUG-ABUSE: Drug abuse, current or past
+2. ALCOHOL-ABUSE: Current alcohol use over weekly recommended limits
+3. ENGLISH: Patient must speak English
+4. MAKES-DECISIONS: Patient must make their own medical decisions
+5. ABDOMINAL: History of intra-abdominal surgery, small or large intestine
+resection, or small bowel obstruction.
+6. MAJOR-DIABETES: Major diabetes-related complication. For the purposes of
+this annotation, we define “major complication” (as opposed to “minor complication”)
+as any of the following that are a result of (or strongly correlated with) uncontrolled diabetes:
+    a. Amputation
+    b. Kidney damage
+    c. Skin conditions
+    d. Retinopathy
+    e. nephropathy
+    f. neuropathy
+7. ADVANCED-CAD: Advanced cardiovascular disease (CAD).
+For the purposes of this annotation, we define “advanced” as having 2 or more of the following:
+    a. Taking 2 or more medications to treat CAD
+    b. History of myocardial infarction (MI)
+    c. Currently experiencing angina
+    d. Ischemia, past or present
+8. MI-6MOS: MI in the past 6 months
+9. KETO-1YR: Diagnosis of ketoacidosis in the past year
+10. DIETSUPP-2MOS: Taken a dietary supplement (excluding vitamin D) in the past 2 months
+11. ASP-FOR-MI: Use of aspirin to prevent MI
+12. HBA1C: Any hemoglobin A1c (HbA1c) value between 6.5% and 9.5%
+13. CREATININE: Serum creatinine > upper limit of normal
+
+The training consists of 202 patient records with document-level annotations, 10 records
+with textual spans indicating annotator’s evidence for their annotations while test set contains 86.
+
+Note: The inter-annotator average agreement is 84.9%
+"""
+
+_HOMEPAGE = "https://portal.dbmi.hms.harvard.edu/projects/n2c2-nlp/"
+
+_LICENSE = "External Data User Agreement"
+
+_SUPPORTED_TASKS = [Tasks.TEXT_CLASSIFICATION]
+
+_SOURCE_VERSION = "1.0.0"
+
+_BIGBIO_VERSION = "1.0.0"
+
+# Constants
+SOURCE = "source"
+BIGBIO_TEXT = "bigbio_text"
+
+
+def _read_zip(file_path, samples=None):
+    if samples is None:
+        samples = defaultdict(dict)
```

Comment:
if `samples` is never passed in, we don't need this logic, we can simply set
`samples = defaultdict(dict)` at the start.

Reply:
true, sorry this method I directly took from your 2011 script. Updated!

Reply:
I recognized it :) in the 2011 script I had to build the samples from two files so needed to have the option to pass in existing samples, but here I think it would just confuse future reviewers.

Reply:
yes, agree 💯